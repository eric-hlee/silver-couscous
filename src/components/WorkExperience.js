import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Card from 'react-bootstrap/Card';
import Carousel from 'react-bootstrap/Carousel';
import ListGroup from 'react-bootstrap/ListGroup';
import Media from 'react-bootstrap/Media';

export default function WorkExperience(props) {
  const [index, setIndex] = useState(2);

  const handleSelect = (selectedIndex) => {
    setIndex(selectedIndex);
  };

  const { workDescriptions } = props;
  return (
    <Carousel activeIndex={index} controls={false} onSelect={handleSelect}>
      <Carousel.Item>
        <Card bsPrefix="card work">
          <Card.Body>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src={require('../assets/images/conversocial.png')}
                alt="Conversocial logo"
              />
              <Media.Body>
                <Card.Title>
                  <Card.Link
                    href="http://www.conversocial.com/"
                    target="_blank"
                    rel="noopener"
                  >
                    <FormattedMessage
                      id="work.conv"
                      defaultMessage="Conversocial"
                    />
                  </Card.Link>
                </Card.Title>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.conv_jse"
                    defaultMessage="Junior Software Engineer"
                  />
                  <span className="info-item">&#183;</span>
                  <FormattedMessage
                    id="work.conv_work_period"
                    defaultMessage="May 2019 – Apr 2020"
                  />
                </Card.Subtitle>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.london_uk"
                    defaultMessage="London, United Kingdom"
                  />
                </Card.Subtitle>
                <ListGroup variant="flush" className="conversocial">
                  {workDescriptions.conversocial &&
                    workDescriptions.conversocial.map((desc, index) => (
                      <ListGroup.Item key={index}>{desc}</ListGroup.Item>
                    ))}
                </ListGroup>
              </Media.Body>
            </Media>
          </Card.Body>
        </Card>
      </Carousel.Item>
      <Carousel.Item>
        <Card bsPrefix="card work">
          <Card.Body>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src={require('../assets/images/verv.jpeg')}
                alt="Verv logo"
              />
              <Media.Body>
                <Card.Title>
                  <Card.Link
                    href="http://verv.energy/"
                    target="_blank"
                    rel="noopener"
                  >
                    <FormattedMessage id="work.verv" defaultMessage="Verv" />
                  </Card.Link>
                </Card.Title>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.verv_tse"
                    defaultMessage="Technical Support Engineer"
                  />
                  <span className="info-item">&#183;</span>
                  <FormattedMessage
                    id="work.verv_work_period"
                    defaultMessage="Oct 2018 – Apr 2019"
                  />
                </Card.Subtitle>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.london_uk"
                    defaultMessage="London, United Kingdom"
                  />
                </Card.Subtitle>
                <ListGroup variant="flush" className="verv">
                  {workDescriptions.verv &&
                    workDescriptions.verv.map((desc, index) => (
                      <ListGroup.Item key={index}>{desc}</ListGroup.Item>
                    ))}
                </ListGroup>
              </Media.Body>
            </Media>
          </Card.Body>
        </Card>
      </Carousel.Item>
      <Carousel.Item>
        <Card bsPrefix="card work">
          <Card.Body>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src={require('../assets/images/ibm.png')}
                alt="IBM logo"
              />
              <Media.Body>
                <Card.Title>
                  <Card.Link
                    href="http://www.ibm.com/"
                    target="_blank"
                    rel="noopener"
                  >
                    <FormattedMessage id="work.ibm" defaultMessage="IBM" />
                  </Card.Link>
                </Card.Title>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.ibm_intern"
                    defaultMessage="e-Commerce Java Development Support Intern"
                  />
                  <span className="info-item">&#183;</span>
                  <FormattedMessage
                    id="work.ibm_work_period"
                    defaultMessage="May 2016 – Apr 2018"
                  />
                </Card.Subtitle>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.toronto_ca"
                    defaultMessage="Toronto, Canada"
                  />
                </Card.Subtitle>
                <ListGroup variant="flush" className="ibm">
                  {workDescriptions.ibm &&
                    workDescriptions.ibm.map((desc, index) => (
                      <ListGroup.Item key={index}>{desc}</ListGroup.Item>
                    ))}
                </ListGroup>
              </Media.Body>
            </Media>
          </Card.Body>
        </Card>
      </Carousel.Item>
      <Carousel.Item>
        <Card bsPrefix="card work">
          <Card.Body>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src={require('../assets/images/queensu.png')}
                alt="Queen's University logo"
              />
              <Media.Body>
                <Card.Title>
                  <Card.Link
                    href="http://www.queensu.ca/"
                    target="_blank"
                    rel="noopener"
                  >
                    <FormattedMessage
                      id="work.queensu"
                      defaultMessage="Queen's University"
                    />
                  </Card.Link>
                </Card.Title>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.queensu_ta"
                    defaultMessage="Teaching Assistant"
                  />
                  <span className="info-item">&#183;</span>
                  <FormattedMessage
                    id="work.queensu_work_period"
                    defaultMessage="Sep 2017 – Apr 2018"
                  />
                </Card.Subtitle>
                <Card.Subtitle className="mb-2 text-muted">
                  <FormattedMessage
                    id="work.kingston_on"
                    defaultMessage="Kingston, Ontario"
                  />
                </Card.Subtitle>
                <ListGroup variant="flush" className="queensu">
                  {workDescriptions.queensu &&
                    workDescriptions.queensu.map((desc, index) => (
                      <ListGroup.Item key={index}>{desc}</ListGroup.Item>
                    ))}
                </ListGroup>
              </Media.Body>
            </Media>
          </Card.Body>
        </Card>
      </Carousel.Item>
    </Carousel>
  );
}

WorkExperience.propTypes = {
  workDescriptions: PropTypes.object.isRequired,
};
