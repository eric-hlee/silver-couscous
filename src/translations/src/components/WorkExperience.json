[{"id":"work.conv","defaultMessage":"Conversocial","file":"src/components/WorkExperience.js","start":{"line":29,"column":10},"end":{"line":32,"column":12}},{"id":"work.conv_jse","defaultMessage":"Junior Software Engineer","file":"src/components/WorkExperience.js","start":{"line":36,"column":9},"end":{"line":39,"column":11}},{"id":"work.conv_work_period","defaultMessage":"May 2019 – Apr 2020","file":"src/components/WorkExperience.js","start":{"line":41,"column":9},"end":{"line":44,"column":11}},{"id":"work.london_uk","defaultMessage":"London, United Kingdom","file":"src/components/WorkExperience.js","start":{"line":92,"column":9},"end":{"line":95,"column":11}},{"id":"work.verv","defaultMessage":"Verv","file":"src/components/WorkExperience.js","start":{"line":77,"column":10},"end":{"line":77,"column":67}},{"id":"work.verv_tse","defaultMessage":"Technical Support Engineer","file":"src/components/WorkExperience.js","start":{"line":81,"column":9},"end":{"line":84,"column":11}},{"id":"work.verv_work_period","defaultMessage":"Oct 2018 – Apr 2019","file":"src/components/WorkExperience.js","start":{"line":86,"column":9},"end":{"line":89,"column":11}},{"id":"work.ibm","defaultMessage":"IBM","file":"src/components/WorkExperience.js","start":{"line":122,"column":10},"end":{"line":122,"column":65}},{"id":"work.ibm_intern","defaultMessage":"e-Commerce Java Development Support Intern","file":"src/components/WorkExperience.js","start":{"line":126,"column":9},"end":{"line":129,"column":11}},{"id":"work.ibm_work_period","defaultMessage":"May 2016 – Apr 2018","file":"src/components/WorkExperience.js","start":{"line":131,"column":9},"end":{"line":134,"column":11}},{"id":"work.toronto_ca","defaultMessage":"Toronto, Canada","file":"src/components/WorkExperience.js","start":{"line":137,"column":9},"end":{"line":140,"column":11}},{"id":"work.queensu","defaultMessage":"Queen's University","file":"src/components/WorkExperience.js","start":{"line":167,"column":10},"end":{"line":170,"column":12}},{"id":"work.queensu_ta","defaultMessage":"Teaching Assistant","file":"src/components/WorkExperience.js","start":{"line":174,"column":9},"end":{"line":177,"column":11}},{"id":"work.queensu_work_period","defaultMessage":"Sep 2017 – Apr 2018","file":"src/components/WorkExperience.js","start":{"line":179,"column":9},"end":{"line":182,"column":11}},{"id":"work.kingston_on","defaultMessage":"Kingston, Ontario","file":"src/components/WorkExperience.js","start":{"line":185,"column":9},"end":{"line":188,"column":11}}]
