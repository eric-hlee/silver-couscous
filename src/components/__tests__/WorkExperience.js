import React from "react";
import { shallow } from "enzyme";
import WorkExperience from "../WorkExperience";

describe("<WorkExperience />", () => {
	it("renders correctly", () => {
		const workDescriptions = {
			conversocial: ["work", "work", "work"],
			verv: ["work", "work", "work"],
			ibm: ["work", "work", "work"],
			queens: ["work", "work", "work"],
		};
		const wrapper = shallow(<WorkExperience workDescriptions />);
		expect(wrapper).toMatchSnapshot();
	});
});
