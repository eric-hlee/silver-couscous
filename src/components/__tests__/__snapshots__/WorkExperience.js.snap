// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkExperience /> renders correctly 1`] = `
<Carousel
  activeIndex={2}
  controls={false}
  defaultActiveIndex={0}
  fade={false}
  indicators={true}
  interval={5000}
  keyboard={true}
  nextIcon={
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
  }
  nextLabel="Next"
  onSelect={[Function]}
  pause="hover"
  prevIcon={
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
  }
  prevLabel="Previous"
  slide={true}
  touch={true}
  wrap={true}
>
  <CarouselItem>
    <Card
      body={false}
      bsPrefix="card work"
    >
      <CardBody>
        <Media>
          <img
            alt="Conversocial logo"
            className="mr-3"
            height={64}
            src="test-file-stub"
            width={64}
          />
          <MediaBody>
            <CardTitle>
              <CardLink
                href="http://www.conversocial.com/"
                rel="noopener"
                target="_blank"
              >
                <FormattedMessage
                  defaultMessage="Conversocial"
                  id="work.conv"
                />
              </CardLink>
            </CardTitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="Junior Software Engineer"
                id="work.conv_jse"
              />
              <span
                className="info-item"
              >
                ·
              </span>
              <FormattedMessage
                defaultMessage="May 2019 – Apr 2020"
                id="work.conv_work_period"
              />
            </CardSubtitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="London, United Kingdom"
                id="work.london_uk"
              />
            </CardSubtitle>
            <ListGroup
              className="conversocial"
              variant="flush"
            >
              <ListGroupItem
                active={false}
                disabled={false}
                key="0"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="1"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="2"
              >
                work
              </ListGroupItem>
            </ListGroup>
          </MediaBody>
        </Media>
      </CardBody>
    </Card>
  </CarouselItem>
  <CarouselItem>
    <Card
      body={false}
      bsPrefix="card work"
    >
      <CardBody>
        <Media>
          <img
            alt="Verv logo"
            className="mr-3"
            height={64}
            src="test-file-stub"
            width={64}
          />
          <MediaBody>
            <CardTitle>
              <CardLink
                href="http://verv.energy/"
                rel="noopener"
                target="_blank"
              >
                <FormattedMessage
                  defaultMessage="Verv"
                  id="work.verv"
                />
              </CardLink>
            </CardTitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="Technical Support Engineer"
                id="work.verv_tse"
              />
              <span
                className="info-item"
              >
                ·
              </span>
              <FormattedMessage
                defaultMessage="Oct 2018 – Apr 2019"
                id="work.verv_work_period"
              />
            </CardSubtitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="London, United Kingdom"
                id="work.london_uk"
              />
            </CardSubtitle>
            <ListGroup
              className="verv"
              variant="flush"
            >
              <ListGroupItem
                active={false}
                disabled={false}
                key="0"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="1"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="2"
              >
                work
              </ListGroupItem>
            </ListGroup>
          </MediaBody>
        </Media>
      </CardBody>
    </Card>
  </CarouselItem>
  <CarouselItem>
    <Card
      body={false}
      bsPrefix="card work"
    >
      <CardBody>
        <Media>
          <img
            alt="IBM logo"
            className="mr-3"
            height={64}
            src="test-file-stub"
            width={64}
          />
          <MediaBody>
            <CardTitle>
              <CardLink
                href="http://www.ibm.com/"
                rel="noopener"
                target="_blank"
              >
                <FormattedMessage
                  defaultMessage="IBM"
                  id="work.ibm"
                />
              </CardLink>
            </CardTitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="e-Commerce Java Development Support Intern"
                id="work.ibm_intern"
              />
              <span
                className="info-item"
              >
                ·
              </span>
              <FormattedMessage
                defaultMessage="May 2016 – Apr 2018"
                id="work.ibm_work_period"
              />
            </CardSubtitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="Toronto, Canada"
                id="work.toronto_ca"
              />
            </CardSubtitle>
            <ListGroup
              className="ibm"
              variant="flush"
            >
              <ListGroupItem
                active={false}
                disabled={false}
                key="0"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="1"
              >
                work
              </ListGroupItem>
              <ListGroupItem
                active={false}
                disabled={false}
                key="2"
              >
                work
              </ListGroupItem>
            </ListGroup>
          </MediaBody>
        </Media>
      </CardBody>
    </Card>
  </CarouselItem>
  <CarouselItem>
    <Card
      body={false}
      bsPrefix="card work"
    >
      <CardBody>
        <Media>
          <img
            alt="Queen's University logo"
            className="mr-3"
            height={64}
            src="test-file-stub"
            width={64}
          />
          <MediaBody>
            <CardTitle>
              <CardLink
                href="http://www.queensu.ca/"
                rel="noopener"
                target="_blank"
              >
                <FormattedMessage
                  defaultMessage="Queen's University"
                  id="work.queensu"
                />
              </CardLink>
            </CardTitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="Teaching Assistant"
                id="work.queensu_ta"
              />
              <span
                className="info-item"
              >
                ·
              </span>
              <FormattedMessage
                defaultMessage="Sep 2017 – Apr 2018"
                id="work.queensu_work_period"
              />
            </CardSubtitle>
            <CardSubtitle
              className="mb-2 text-muted"
            >
              <FormattedMessage
                defaultMessage="Kingston, Ontario"
                id="work.kingston_on"
              />
            </CardSubtitle>
            <ListGroup
              className="queensu"
              variant="flush"
            />
          </MediaBody>
        </Media>
      </CardBody>
    </Card>
  </CarouselItem>
</Carousel>
`;
